### PR TITLE
Fix individual crate compilation issues & add script for running all tests

### LINKED ISSUE #1441: Script to check and test all crates separately
### Description

When testing the entire workspace at once, rust chooses the crates with more features on them. This means that individually our crates can be broken, but because we are compiling all together it just works. 

We can avoid this situation and be able to work faster compiling and testing each crate separately with any combination of features as follows:

```sh
cargo test -p <my-crate> -F runtime-benchmarks
```

Also, having them isolated compiling helps to find general compilation issues due to features in the future.

### Propose
- Leave all crates without any compilation issues.
- Add a job to check automatically that all crates work separately with all feature combinations. This can take time, so I would propose creating a nightly job to do this.

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Fixes #1441 

## Changes and Descriptions

- Fixed all compilation issues running all crates isolated 🎉. Now you can run without errors each crate as:
  ```sh
  cargo test -p <crate> -F <feature1,feature2,...>
  ``` 
- Added script to run all workspace crate tests with all feature combinations.
 
  To run the script for all crates:
  ```sh
  ./scripts/tests.sh
  ```

  To run the script for all crates from the selected one in the workspace list (used mainly to recover the script execution point after a crate failure):
  ```sh
  ./scripts/tests.sh <from-crate>
  ```

### CODE REVIEW

File: runtime/common/Cargo.toml
Comment (lemunozm) [inline runtime/common/Cargo.toml:47]: I'm pretty sure this was the issue that happened in https://github.com/centrifuge/centrifuge-chain/pull/1406 (not tested yet), so it unlocks that PR.

File: pallets/migration/src/benchmarking.rs
Comment (lemunozm) [inline pallets/migration/src/benchmarking.rs:195]: This allows us to remove the stack limit increment in the CI scripts

**Review by lemunozm — COMMENTED** [review]
A couple of "relevant" fixes comes with this PR:

**Comment by lemunozm** [comment]
It would be great to add this script as a CI job, but it surely exceeds the maximum size available and will take considerably more time than other jobs. Any idea @gpmayorga? My suggestion is to clean artifacts after testing each crate (which will take even more time to run all crates) and create a weekend job for it.

File: runtime/common/Cargo.toml
Comment (NunoAlexandre) [reply runtime/common/Cargo.toml:47]: That could very well be it. But I tried using my fork of frontier fixing all the cargo features and the issue still persisted 🤔

File: scripts/tests.sh
Comment (NunoAlexandre) [inline scripts/tests.sh:36]: Can we have this as part of a `setup` script? Having this here means that every time we want to run this (locally) we will have cargo checking for updates and trying to update this crate, which will slow this process even further. wdyt?

File: runtime/common/Cargo.toml
Comment (lemunozm) [reply runtime/common/Cargo.toml:47]: Have you added `"fp-self-contained/try-runtime"` too to your try?

File: scripts/tests.sh
Comment (lemunozm) [reply scripts/tests.sh:36]: Sure! Make sense. What's the `setup` script you're referring?

File: runtime/common/Cargo.toml
Comment (NunoAlexandre) [reply runtime/common/Cargo.toml:47]: Not sure, but I ran subalfred on the whole repo to make sure I wasn't missing any feature anywhere

File: scripts/tests.sh
Comment (NunoAlexandre) [reply scripts/tests.sh:36]: It doesn't quite exist atm, but we could create one or adapt the current install-toolchain to be more general-purpose and install tools we need for development

File: runtime/common/Cargo.toml
Comment (lemunozm) [reply runtime/common/Cargo.toml:47]: Ah, but before this PR I also run `subalfred` for all crates, and everything went "well" 😆, and things weren't well. I guess we can not trust 100% in `subalfred`. Some things are not checked.

File: scripts/tests.sh
Comment (lemunozm) [reply scripts/tests.sh]: Ok, I understand. By now, f you agree, I can print a message if the command is not found instead of fetching every time if it exists. I do not want to add things into `install_toolchain.sh` because it affects all CI job time/space. Only the required job should fetch its required tools, I think.

Comment (lemunozm) [reply scripts/tests.sh]: We can prepare a good script in another PR, with taplo, subalfred, cargo-workspaces, ...

Comment (NunoAlexandre) [reply scripts/tests.sh]: Yeah that's fine :)

**Review by NunoAlexandre — APPROVED** [review]
LGTM  but it would be nice to have another reviewer 👍🏻

**Comment by lemunozm** [comment]
Thanks Nuno! I think other review is even required 🤣

**Review by mustermeiszer — APPROVED** [review]
LGTM!

File: pallets/pool-registry/src/mock.rs
Comment (branan) [inline pallets/pool-registry/src/mock.rs:226]: Are we sure the only code paths for this can be thru the runtime-benchmarks feature? I'm a little concerned about having the mocks behave differently between benches and tests...

**Review by branan — APPROVED** [review]
One question / minor concern, but it's not a blocker. Leaving my approval so merging isn't blocked on the discussion

**Review by wischli — APPROVED** [review]
LGTM!

File: pallets/pool-registry/src/mock.rs
Comment (lemunozm) [reply pallets/pool-registry/src/mock.rs:226]: Good point! It's true that it's used in tests even when `runtime-benchmarks` is not enabled, so in that case, you're not creating a pool. Anyways, it is compiling and passing with and without `runtime-benchmarks` feature, so I suppose it should behave as expected.

Looking more deeply creating a pool seems to doesn't matter for the testing, but yes for the benchmarks because it adds a lot of weight to the call, so makes sense to only call `create_pool()` on benchmaking.

File: pallets/migration/src/benchmarking.rs
Comment (mikiquantum) [reply pallets/migration/src/benchmarking.rs:195]: I think we should remove this pallet all-together. We will not use it again.

**Review by mikiquantum — APPROVED** [review]
I would remove the migrations pallet, but in any case I am not blocking this pr 👍

File: pallets/migration/src/benchmarking.rs
Comment (lemunozm) [reply pallets/migration/src/benchmarking.rs:195]: Good to know!

Comment (lemunozm) [reply pallets/migration/src/benchmarking.rs:195]: https://github.com/centrifuge/centrifuge-chain/issues/1448

**Pull request merged**
